python/python/lance/dataset.py

```diff
@@ -1603,8 +1603,10 @@ def write_dataset(
     uri: Union[str, Path],
     schema: Optional[pa.Schema] = None,
     mode: str = "create",
+    *,
```

Comment:
Given we have supported it for a while, I think it's reasonable we support `schema` and `mode` being passed positionally. But the other ones that are integers and boolean options are difficult to read if they aren't passed as keyword arguments, so that's why I start there.